api/rosetta/balance.go

```diff
@@ -58,12 +58,19 @@ func (d *Data) Balance(ctx echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(networkEmpty))
 	}
 
-	if req.BlockID.Index == nil && req.BlockID.Hash == "" {
-		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(blockEmpty))
+	rosBlockID := req.BlockID
```

Comment:
We mentioned the option of having the validator infer this. I thought this might be a more direct method and is clearer when reading the handler what is being requested.

If needed I could change it and go the validator route. The changes for the tests should remain applicable.